src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+    Math_SmoothStepToF(&player->camAt.x, D_801779A0, D_80177A48[0], 50000.0f, 0);
+    Math_SmoothStepToF(&player->camAt.y, D_801779B8, D_80177A48[0], 50000.0f, 0);
+    Math_SmoothStepToF(&player->camAt.z, D_801779C0, D_80177A48[0], 50000.0f, 0);
+
+    Matrix_RotateY(gCalcMatrix, (player->unk_0E8 + 180.0f) * M_DTOR, 0);
+    Matrix_RotateX(gCalcMatrix, -((player->unk_0E4 + player->unk_4D8) * M_DTOR), 1);
+
+    src.x = 0.0f;
+    src.y = 0.0f;
+    src.z = player->unk_0D0 + player->unk_110;
+
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+
+    player->vel.x = dest.x;
+    player->vel.z = dest.z;
+    player->vel.y = dest.y;
+
+    player->pos.x += player->vel.x;
+    player->pos.y += player->vel.y;
+    player->pos.z += player->vel.z;
+
+    player->unk_138 = player->pos.z + player->unk_08C;
+    player->unk_0F8 = player->unk_0EC + player->unk_12C + player->unk_130;
+    player->unk_088 += 10.0f;
+    player->unk_080 = -__sinf(player->unk_088 * M_DTOR) * 0.3f;
+    player->unk_0F4 += 8.0f;
+    player->unk_0F0 = __sinf(player->unk_0F4 * M_DTOR);
+}
+
+void func_i2_8018ED9C(Actor* actor) {
+    Vec3f sp3C;
+    Vec3f sp30;
+
+    switch (actor->unk_0B8) {
+        case 0:
+            actor->vwork[0].x = D_i2_80195610[actor->index].x + gPlayer->pos.x;
+            actor->vwork[0].y = D_i2_80195610[actor->index].y + gPlayer->pos.y;
+            actor->vwork[0].z = D_i2_80195610[actor->index].z + gPlayer->pos.z;
+
+            Math_SmoothStepToF(&actor->obj.pos.x, actor->vwork[0].x, 0.02f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&actor->obj.pos.y, actor->vwork[0].y, 0.02f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&actor->obj.pos.z, actor->vwork[0].z, 0.02f, 50.0f, 0.0001f);
+            Math_SmoothStepToF(&actor->unk_0F4.z, 0.0f, 0.03f, 0.5f, 0.0001f);
+            break;
+
+        case 1:
+            actor->unk_0B8 = 2;
+            Audio_PlaySfx(0x09000002U, &actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            actor->timer_0BC = 0x96;
+            actor->fwork[29] = 5.0f;
+
+        case 2:
+            actor->iwork[11] = 2;
+            actor->fwork[0] += 2.0f;
+            actor->unk_0F4.x += 0.1f;
+            if (actor->timer_0BC == 0) {
+                Object_Kill(&actor->obj, &actor->sfxPos);
+            }
+            break;
+    }
+
+    Matrix_RotateY(gCalcMatrix, (actor->unk_0F4.y + 180.0f) * M_DTOR, 0U);
+    Matrix_RotateX(gCalcMatrix, -(actor->unk_0F4.x * M_DTOR), 1U);
+
+    sp3C.x = 0.0f;
+    sp3C.y = 0.0f;
+    sp3C.z = actor->fwork[0];
+
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp3C, &sp30);
+
+    actor->vel.x = sp30.x;
+    actor->vel.y = sp30.y;
+    actor->vel.z = sp30.z;
+
+    actor->obj.rot.x = -actor->unk_0F4.x;
+    actor->obj.rot.y = actor->unk_0F4.y + 180.0f;
+    actor->obj.rot.z = -actor->unk_0F4.z;
+}
 
 #pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/D_i2_80195790.s")
```

Comment:
`"Demo_Time %d\n"` refer to `D_80177A80`. It probably goes near one of the switches on that variable.